fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/center/CenterImportHandler.java

```diff
@@ -0,0 +1,288 @@
+        for (int cellNo =CenterConstants. GROUP_NAMES_STARTING_COL; cellNo < CenterConstants.GROUP_NAMES_ENDING_COL; cellNo++) {
+            String groupName = ImportHandlerUtils.readAsString(cellNo, row);
+            if (groupName==null||groupName.equals(""))
+                break;
+            Long groupId = ImportHandlerUtils.getIdByName(workbook.getSheet(TemplatePopulateImportConstants.GROUP_SHEET_NAME), groupName);
+            GroupGeneralData group = new GroupGeneralData(groupId);
+            if (!containsGroupId(groupMembers,groupId)) {
+                groupMembers.add(group);
+            }
+        }
+
+        statuses.add(status);
+        return CenterData.importInstance(centerName,groupMembers,activationDate, active,submittedOn, externalId,
+                officeId, staffId, row.getRowNum(),dateFormat,locale);
+    }
+
+    private boolean containsGroupId(List<GroupGeneralData> groupMembers,Long groupId){
+        for (GroupGeneralData group: groupMembers) {
+            if (group.getId()==groupId){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public Count importEntity(String dateFormat) {
+        Sheet centerSheet = workbook.getSheet(TemplatePopulateImportConstants.CENTER_SHEET_NAME);
+        int progressLevel = 0;
+        String centerId = "";
+        int successCount = 0;
+        int errorCount = 0;
+        String errorMessage = "";
+        for (int i = 0; i < centers.size(); i++) {
+            Row row = centerSheet.getRow(centers.get(i).getRowIndex());
+            Cell errorReportCell = row.createCell(CenterConstants.FAILURE_COL);
+            Cell statusCell = row.createCell(CenterConstants.STATUS_COL);
+            CommandProcessingResult result = null;
+            try {
+                String status = statuses.get(i);
+                progressLevel = getProgressLevel(status);
+
+                if (progressLevel == 0) {
+                    result = importCenter(i, dateFormat);
+                    centerId = result.getGroupId().toString();
+                    progressLevel = 1;
+                } else
+                    centerId = ImportHandlerUtils.readAsInt(CenterConstants.CENTER_ID_COL, centerSheet.getRow(centers.get(i).getRowIndex())).toString();
+
+                if (meetings.get(i) != null)
+                    progressLevel = importCenterMeeting(result, i, dateFormat);
+                successCount++;
+                statusCell.setCellValue(TemplatePopulateImportConstants.STATUS_CELL_IMPORTED);
+                statusCell.setCellStyle(ImportHandlerUtils.getCellStyle(workbook, IndexedColors.LIGHT_GREEN));
+            } catch (AbstractPlatformDomainRuleException e) {
+                errorCount++;
+                e.printStackTrace();
+                errorMessage = e.getDefaultUserMessage();
+                writeCenterErrorMessage(centerId,errorMessage,progressLevel,statusCell,errorReportCell,row);
```

Comment:
Move all the catch blocks to a functions in your Utils class and call it to get the error message. Call this function from all your import handler classes.